src/CompletionHandler.php

```diff
@@ -442,4 +441,19 @@ protected function getAllOptions()
             $this->application->getDefinition()->getOptions()
         );
     }
+
+    protected function getCommandNames()
+    {
+        $commands = array();
+
+        foreach ($this->application->all() as $name => $command) {
+            if (($this->commandsHidable and $command->isHidden()) or (!$this->commandsHidable and $name === '_completion')) {
```

Comment:
We'd better make 2 foreach inside if/else and this way we won't get speed decrease while checking for symfony version for each of processed commands.